### ISSUE TITLE
Container state "Removal In Progress" when ntpd/chronyd is active

### ISSUE THREAD

**Issue by victorvarza** [docker/for-linux#124]
* [x] This is a bug report
* [ ] This is a feature request
* [x] I searched existing issues before opening this one

### Expected behavior
Docker remove container when doing "docker rm container_name"

### Actual behavior
Container hang into "Removal in Progress" State. Actually docker daemon do a loop by trying remove the  container using: "docker rm container_name". Every time it exists with code error 1. 

### Steps to reproduce the behavior

a) Run any image (I noticed on some images like: haproxy, uchiwa, rabbitmq, but I think it is applied to any image, not 100% sure). 
b) Wait 1-2 days
c) Try remove the container using "docker rm container_name"
d) Container will be removed only after ntpd/chronyd will be restarted

**Output of `docker version`:**

```
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:06:25 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:06:25 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 7
 Running: 7
 Paused: 0
 Stopped: 0
Images: 70
Server Version: 17.05.0-ce
Storage Driver: devicemapper
 Pool Name: docker-202:2-1033721-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 107.4GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 16.37GB
 Data Space Total: 107.4GB
 Data Space Available: 68.43GB
 Metadata Space Used: 32.39MB
 Metadata Space Total: 2.147GB
 Metadata Space Available: 2.115GB
 Thin Pool Minimum Free Space: 10.74GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.140-RHEL7 (2017-05-03)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9048e5e50717ea4497b757314bad98ea3763c145
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-693.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 31.26GiB
Name: my_server
ID: my_id
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Http Proxy: http://internal_proxy:3128
Https Proxy: http://internal_proxy:3128
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 internal_registry:443
 127.0.0.0/8
Live Restore Enabled: false

WARNING: devicemapper: usage of loopback devices is strongly discouraged for production use.
         Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.

```

**Additional environment details (AWS, VirtualBox, physical, etc.)**
Docker runs on a VM in SoftLayer cloud.
VM details:
#cat /etc/redhat-release 
CentOS Linux release 7.4.1708 (Core) 
#uname   -r
3.10.0-693.2.2.el7.x86_64
#ntpd --version 
ntpd 4.2.6p5

I've attached strace files when trying to remove the container and exit is 1. 
The second file is when trying to remove the container and restart in the same time the ntpd daemon and then the exit is 0:
strace -f -s 2048 -p  $(ps -ef | grep uchiwa |  grep -v grep | tr -s ' '  | cut -d ' ' -f2)

[exit_0_ntpd_restarted.log](https://github.com/docker/for-linux/files/1368002/exit_0_ntpd_restarted.log)
[exit_1.log](https://github.com/docker/for-linux/files/1368001/exit_1.log)

**Comment by yunghoy** [comment]
Around 1500 of the same "Removal In Progress" containers use all swarm cluster resources.
Swarm "task-history-limit" is set to 1.

**Comment by yunghoy** [comment]
It happens every Monday. See victorvarza's commants, "Wait 1-2 days."
If it happens, "Removal In Progress" state always happens all the time even after pruning and restarting docker.

**Comment by yunghoy** [comment]
After deleting 2000*6(12000) "Removal In Progress" state containers, I've restarted a small service.

$ docker stack ls
NAME                SERVICES
~~~              42
$ docker ps -a | grep Removal | wc -l
45

Really?
I have got to send "echo y | docker container prune" command to all nodes with pssh.
Can I use only docker command instead of using pssh command, please?
Swarm mode is so fragile and buggy I cannot even think of using it in live service.
Should I go back to Docker Swarm?

**Comment by yunghoy** [comment]
After restarting a service several times, "Removal In Progress" state has been disappeared. It may happen again next Monday. Let me comment again on the thread next Monday.

**Comment by cpuguy83** [comment]
I'm curious about the state being stuck in `Removal In Progress`.
Can you get a goroutine dump with his happens so we can check if its still just actually trying to remove and stuck somewhere?

You may also want to try setting `echo 1 > /proc/sys/fs/may_detach_mounts` and see if that clears up your problems. This only works on CentOS/RHEL 7.4 and is set by default in 17.09 (latest patch release) and 17.10.

**Comment by AaronDMarasco-VSI** [comment]
I just ran into the same problem. On my Jenkins server, I had 39 containers stuck in the "Removal" process starting some time last night. I had a handful of "docker inspect" calls hung as well.

If I restarted `ntpd`, nothing changed. When I straight-out stopped `ntpd`, it looks like _everything_ died and all my containers went away:
```
00:02:23.793 [Build PDF Documentation] + docker inspect -f . jenkins/ocpi_pdfgen:v2
00:17:30.047 [Build PDF Documentation] Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

Output of `ps`: [docker.log](https://github.com/docker/for-linux/files/1440706/docker.log)

Docker version 17.09.0-ce, build afdb6d4, running on CentOS 7.3.1611 (3.10.0-514.6.1.el7.x86_64). I don't (yet) have `/proc/sys/fs/may_detach_mounts` that @cpuguy83 mentioned.

**Comment by victorvarza** [comment]
@cpuguy83 thanks for your suggestion. I've tried with "echo 1 > /proc/sys/fs/may_detach_mounts" but unfortunately didn't work. 
The strace dumps are attached in my first comment. The exist_1.log is the file where docker continuously trying to remove a container all over again without success.

**Comment by cpuguy83** [comment]
@victorvarza unfortunately strace isn't really helpful here, but docekr can dump it's full goroutine stack if you send `SIGUSR1` to the dockerd process, or hit `/debug/pprof/goroutine?debug=2`

**Comment by victorvarza** [comment]
I added goroutine stack by sending SIGUSR1 to docker daemon. 
At the moment of dump I had 2 containers in 'Removal in progress' state, and 4 running. I didn't mentioned that the containers are started via systemd, as a separate unit. 

[goroutine-stacks-2017-11-08T210103Z.log](https://github.com/docker/for-linux/files/1455433/goroutine-stacks-2017-11-08T210103Z.log)

**Comment by cpuguy83** [comment]
@victorvarza Is this still 17.05?

**Comment by victorvarza** [comment]
Yes, docker version is 17.05.0-ce

**Comment by cpuguy83** [comment]
@victorvarza Can you upgrade to a more recent version like 17.09? Pretty sure you are hitting a bug in 17.05.

Indeed in your stack dump it's stuck calling `devicemapper.RemoveDevice()` which had some issues.

**Comment by cpuguy83** [comment]
I think 17.06.2 would have this fix as well I think, but 17.06 is EOL.

**Comment by victorvarza** [comment]
I will upgrade it and I will let you know the results.

**Comment by AaronDMarasco-VSI** [comment]
I'm still hitting this [in 17.09](https://github.com/docker/for-linux/issues/124#issuecomment-341666868)...

**Comment by cpuguy83** [comment]
@AaronDMarasco-VSI Can you provide goroutine stack dump?

**Comment by cpuguy83** [comment]
And `docker info` output

**Comment by AaronDMarasco-VSI** [comment]
I can give you the latter. Right now, there are none in that state because I restarted docker this morning. If it comes up again, I'll search how to do it.
```
$ docker info
Containers: 130
 Running: 8
 Paused: 0
 Stopped: 122
Images: 263
Server Version: 17.09.0-ce
Storage Driver: devicemapper
 Pool Name: vg_ex-docker--pool
 Pool Blocksize: 65.54kB
 Base Device Size: 107.4GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 679.9GB
 Data Space Total: 1.1TB
 Data Space Available: 419.6GB
 Metadata Space Used: 611.9MB
 Metadata Space Total: 4.295GB
 Metadata Space Available: 3.683GB
 Thin Pool Minimum Free Space: 110GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.6.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 56
Total Memory: 251.6GiB
Name: XXX
ID: Z7J7:CLEG:KZPK:TNEI:QLYL:JBTO:XNM4:NX2X:KPQC:VHPC:6SFS:G4GR
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by sweharris** [comment]
FWIW, I'm seeing this issue with 17.09

```
CONTAINER ID        IMAGE                             COMMAND                  CREATED             STATUS                 PORTS                    NAMES
84e9a5d43c07        dockersamples/visualizer:latest   "npm start"              3 hours ago         Up 3 hours (healthy)   8080/tcp                 viz_viz.1.so2z72fla62hvvhhq37hrgyck
7c2e586e58ca        postgres:9.6                      "docker-entrypoint..."   3 hours ago         Up 3 hours             5432/tcp                 clair_postgres.1.i9q9fjx9friqyud4y6eslhj2g
ce0e2c6e0d50        quay.io/coreos/clair:latest       "/clair -config=/c..."   3 hours ago         Up 3 hours             6060-6061/tcp            clair_clair.1.1ohx8rojl9x8fkoqdtcdasx7t
b84ad4061a37        portainer/portainer               "/portainer"             3 hours ago         Up 3 hours             0.0.0.0:9000->9000/tcp   elastic_hodgkin
850fb4d3daf3        postgres:9.6                      "docker-entrypoint..."   3 weeks ago         Removal In Progress                             clair_postgres.1.2tu8rgt7ctmlbfacrjlgkqsq7

```
I see this in dmesg output:

    [1561626.492804] device-mapper: thin: Deletion of thin device 111 failed.

The thin device definitely doesn't exist
```
% dmsetup status | grep docker
docker-253:9-360449-9ca9a7d78430705bbaa55d308a000466585ecfac824e97d771758666bfbc89b7: 0 20971520 thin 605184 20971519
docker-253:9-360449-pool: 0 41943040 thin-pool 489 1518/262144 49038/327680 - rw no_discard_passdown queue_if_no_space -
docker-253:9-360449-0e4094b55a6b377f8887619d98c1628a6512a097712b45c0e1a8088da8808982: 0 20971520 thin 446592 20971519
appvg-docker--data: 0 20971520 linear
appvg-docker--data: 20971520 20971520 linear
docker-253:9-360449-9520f300e0f53c353f45d56a09060a0382f2ae17bb0fe1687345999b483b5fb1: 0 20971520 thin 782080 20971519
appvg-docker--meta: 0 2097152 linear
docker-253:9-360449-a451629b4ca8b2c1839c17d6b038216df974dd08c9ffec6395289258cef964ff: 0 20971520 thin 89088 20971519
appvg-docker--fs: 0 20971520 linear


% docker rm 850fb4d3daf3
Error response from daemon: driver "devicemapper" failed to remove root filesystem for 850fb4d3daf334489892d81a99bd37717189c3cc5e41c3a42d845f83af4e833e: failed to remove device ed0ec0e4028e44d22d0876f4c213157565671f95a12e4488fd4841f8fed7b5bf: devicemapper: Error running DeleteDevice dm_task_run failed
```

I've attached a stack log if it'll help.  The only way I've found to clear this is to stop dockerd and clean out /var/lib/docker :-(

[goroutine-stacks-2017-11-14T180628Z.log.gz](https://github.com/docker/for-linux/files/1472040/goroutine-stacks-2017-11-14T180628Z.log.gz)

**Comment by amerenda** [comment]
I'm also seeing this issue with 17.09.0-ce


`$ docker ps -a  | grep "Removal In Progress" | awk '{ print $1 }' | xargs -L 1 docker rm
Error response from daemon: driver "devicemapper" failed to remove root filesystem for ffa030d50d6879295906617d0e256313c2edd89978663a86b4bc0179c4a5fe61: remove /var/lib/docker/devicemapper/mnt/de95c5e24296724be05d581f17a2521f01fb6a7fae1562ba8eacf63aab039a07: device or resource busy
Error response from daemon: driver "devicemapper" failed to remove root filesystem for 27a2c3834a3b5faddcd003882716f14898ebafa5f29c0c0ab2f8dbf68433f1a6: remove /var/lib/docker/devicemapper/mnt/64a14a4df92d5a090f64a0647119911c2551a869cf7520582478dd1ab82bd2fd: device or resource busy`


I found that a few processes are keeping the device mapper busy:

`$ MOUNT=64a14a4df92d5a090f64a0647119911c2551a869cf7520582478dd1ab82bd2fd`
`$ find /proc/*/mounts | xargs grep $MOUNT 2>/dev/null | cut -d/ -f3 | xargs -L 1 ps -q | awk '{ print $4 }' | grep -v CMD
ntpd
nginx
nginx
nginx
nginx
nginx
nginx
nginx
nginx
nginx`


If I restart those services, the containers will be deleted. 

I've attached the output of my docker info. 

[dockerInfo.txt](https://github.com/docker/for-linux/files/1564175/dockerInfo.txt)

**Comment by jaredkosanovic** [comment]
I'm also seeing this issue with 17.09.1-ce on Debian Jessie. I'm able to manaully remove the containers that are stuck in `Removal In Progress` simply with `docker rm` (no -f).

Output of docker info:
```
Containers: 7
 Running: 0
 Paused: 0
 Stopped: 7
Images: 298
Server Version: 17.09.1-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 1973
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.7GiB
Name: my-vm
ID: LBMF:LMQ6:DAHG:X5P6:ZOJA:XQKT:ZLJF:POWW:3EZN:7Q4F:73MY:ERCX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

At the time the containers are first attempted to be removed (which throws them into "Removal In Progress" limbo), these messages are given in /var/log/daemon.log 
```
Dec 25 08:30:05 my-vm dockerd[64080]: time="2017-12-25T08:30:05.579601443-08:00" level=error msg="Error removing mounted layer 5a7756c7d1677de2f8d1508b0f28316e703f1731
5d2899308e1bd831dc2df880: could not remove diff path for id 296bd0784e832cc2d890993acb0e5925c52d5a8db8e9a514433519b9722ebcca: error preparing atomic delete: rename /var/lib/
docker/aufs/diff/296bd0784e832cc2d890993acb0e5925c52d5a8db8e9a514433519b9722ebcca /var/lib/docker/aufs/diff/296bd0784e832cc2d890993acb0e5925c52d5a8db8e9a514433519b9722ebcca-
removing: device or resource busy"
Dec 25 08:30:05 my-vm dockerd[64080]: time="2017-12-25T08:30:05.579763634-08:00" level=error msg="error removing container" container=5a7756c7d1677de2f8d1508b0f28316e7
03f17315d2899308e1bd831dc2df880 error="driver \"aufs\" failed to remove root filesystem for 5a7756c7d1677de2f8d1508b0f28316e703f17315d2899308e1bd831dc2df880: could not remov
e diff path for id 296bd0784e832cc2d890993acb0e5925c52d5a8db8e9a514433519b9722ebcca: error preparing atomic delete: rename /var/lib/docker/aufs/diff/296bd0784e832cc2d890993a
cb0e5925c52d5a8db8e9a514433519b9722ebcca /var/lib/docker/aufs/diff/296bd0784e832cc2d890993acb0e5925c52d5a8db8e9a514433519b9722ebcca-removing: device or resource busy"
```

Attached goroutine dump while one container was stuck in "Removal In Progress".
[goroutine-stacks-2017-12-27T080422-0800.log](https://github.com/docker/for-linux/files/1589447/goroutine-stacks-2017-12-27T080422-0800.log)

**Comment by rebel24** [comment]
This is also happening to me on centos 7.4 

Any help would be great.

**Comment by gdhgdhgdh** [comment]
Same - RHEL 7.3 with Docker CE 17.09.1 and devicemapper thin pools. We're almost at the stage of cronning a task to look for `Waiting for Removal` and `docker rm $(docker ps -aq)` if it finds any. :(

New goroutine-stacks dump attached: [stacks.txt](https://github.com/docker/for-linux/files/1710277/a.txt)

**Comment by flyhard** [comment]
Seeing it in docker 17.12.0-ce

```
Containers: 5
 Running: 5
 Paused: 0
 Stopped: 0
Images: 17
Server Version: 17.12.0-ce
Storage Driver: devicemapper
 Pool Name: docker-253:3-201328978-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Udev Sync Supported: true
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Data Space Used: 5.667GB
 Data Space Total: 107.4GB
 Data Space Available: 31.57GB
 Metadata Space Used: 8.962MB
 Metadata Space Total: 2.147GB
 Metadata Space Available: 2.139GB
 Thin Pool Minimum Free Space: 10.74GB
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.140-RHEL7 (2017-05-03)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 89623f28b87a6004d4b785663257362d1658a729
runc version: b2567b37d7b75eb4cf325b77297b140ea686ce8f
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.10.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 62.72GiB
Name: wa1
ID: LOTK:VXZZ:GKWA:NK4N:FZFN:DGJR:JMGA:3VFU:ZZLC:FWII:FJPP:7KWX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by adidenko** [comment]
I have the same issue on CentOS 7 nodes with kernel 3.10. The only node that does not have this problem runs kernel 4.4.

**Comment by cpuguy83** [comment]
This should all be fixed in 17.12.1 and 18.03 regardless of what kernel/distro you are using.

**Comment by cpuguy83** [comment]
Please test and let us know.
I'm going to close this as resolved by the following PR's:

- https://github.com/moby/moby/pull/36096
- https://github.com/moby/moby/pull/36055
- https://github.com/moby/moby/pull/36047

If you still have issues on 17.12.1 or 18.03, please let us know!

**Comment by sweharris** [comment]
I'm still having issues

```
% docker version
Client:
 Version:       17.12.1-ce
 API version:   1.35
 Go version:    go1.9.4
 Git commit:    7390fc6
 Built: Tue Feb 27 22:15:20 2018
 OS/Arch:       linux/amd64

Server:
 Engine:
  Version:      17.12.1-ce
  API version:  1.35 (minimum version 1.12)
  Go version:   go1.9.4
  Git commit:   7390fc6
  Built:        Tue Feb 27 22:17:54 2018
  OS/Arch:      linux/amd64
  Experimental: false

% docker ps -a
CONTAINER ID        IMAGE                             COMMAND                  CREATED             STATUS              PORTS               NAMES
...
8119afc499d0        dockersamples/visualizer:latest   "npm start"              2 weeks ago         Dead                                    viz_viz.1.2ok0e3ioiifh5ttfedgk1tp9w

% docker rm 8119afc499d0
Error response from daemon: driver "devicemapper" failed to remove root filesystem for 8119afc499d0e78cd9441f37be239eb28a6b8176ffc4e83927519e81a0728b8f: failed to remove device 4f92f556c1a9fceb63f93d5841c4f98be5a74f43c5e64b326a11e20d4b8d6c08-init: devicemapper: Error running DeleteDevice dm_task_run failed

% docker ps -a
...
8119afc499d0        dockersamples/visualizer:latest   "npm start"              2 weeks ago         Removal In Progress                       viz_viz.1.2ok0e3ioiifh5ttfedgk1tp9w
```

**Comment by adidenko** [comment]
OK, upgrade to 17.12.1-ce has fixes the issue for me. Tested on k8s cluster with two exactly the same nodes (CentOS) - one has docker 17.12.1-ce (no problems now), the other has 17.09.1-ce (containers in "Removal in progress" state).

**Comment by jshort** [comment]
I see this on OL7 and only on certain hosts in our pool.  It's always ntpd that's holding on that prevents the container from being deleted (puts container in DEAD state).  Is there a workaround if we will be on 17.06.2-ol for a while?

**Comment by wpeng102** [comment]
@cpuguy83,  Our customer also meet `device or resource busy` issue in 17.09.0-ce, is there any workaround for issue in issue in 17.09.0-ce? 

```
Apr 10 19:01:26 s00vl9984067 dockerd[2162]: time="2018-04-10T19:01:26.517481867+02:00" level=error msg="Handler for DELETE /v1.31/containers/28a61840350dffdbb579cf971dece93ece7493a3eaff1fa51210f9718d2c9dd6 returned error: unable to remove filesystem for 28a61840350dffdbb579cf971dece93ece7493a3eaff1fa51210f9718d2c9dd6: remove /var/lib/docker/containers/28a61840350dffdbb579cf971dece93ece7493a3eaff1fa51210f9718d2c9dd6/shm: device or resource busy"
```

**Comment by cpuguy83** [comment]
Use "rslave" for any containers that are mounting in dirs from the docker
root (or any of its parent dirs).

**Comment by oliveiradouglas** [comment]
Restarting the docker service work for me.

`sudo docker service restart`
